### ISSUE TITLE
QuickFolders toolbar preview mode bug

### ISSUE THREAD

**Issue by burval** [RealRaven2000/QuickFolders#557]
1) QuickFolders toolbar "preview mode" (added in 6.10) remains enabled after closing QuickFolders Options window.

![Image](https://github.com/user-attachments/assets/c6feffa9-4a86-4670-935b-89bbe0d5f536)

2) QuickFolders toolbar "preview mode" appears to be enabled at startup (rather than disabled)

**Comment by RealRaven2000** [comment]
interesting, do you recall what other tabs were open or which tab you changed to after closing options? Maybe there were no tabs left and the tabbar was hidden?

**Comment by RealRaven2000** [comment]
So far I am not able to reproduce the issue (especially to make it visible when starting Thunderbird) I guess that QuickFolders options was active when you started Thunderbird?

in my own system, when I change to the 3pane tab or when I close options tab and the system changes to it, the "preview mode" is being removed automatically. are you using multiple Thunderbird windows maybe?

By default this element is invisible when it is injected into the toolbar:

```xhtml
        <div id="QuickFolders-toolPanel" align="center" style="display: inline-flex; align-items: stretch !important;">
          <vbox id="QuickFolders-LabelBox" flex="0">
            <toolbarbutton id="QuickFolders-title-label" 
              label="__MSG_qf.label.quickfolders__" 
              tabindex="0"/>
            <span id="QuickFolders-PreviewToolbar" visible="false">(__MSG_preview__)</span>
          </vbox>
        </div>
```

**Comment by RealRaven2000** [comment]
OK, I know what happened: I was using the new option "Always display options in a tab" and this would only affect the "demo" version of the toolbar on that tab. Eventually this option will be switched on by default, you can enable it as a workaround:

![Image](https://github.com/user-attachments/assets/edde77b8-c1b0-44d3-9dc2-f99a72677b03)

Should be easy enough to fix I hope... 
in the meantime you can also remove the label by clicking on another tab and then clicking on the main tab again.

**Comment by burval** [comment]
Impressive turnaround time @RealRaven2000 and thanks for the workaround.  I already hold a Pro license and renew annually without hesitation but would like to make an additional donation, please advise on how I can do so.

**Comment by RealRaven2000** [comment]
Quick fix:

[QuickFolders-mx-6.10.1pre45.zip](https://github.com/user-attachments/files/19353274/QuickFolders-mx-6.10.1pre45.zip)

this was a tricky one! apparently the tab listener in the background script is also triggered when I open the options in a dialog window (internally this is still handled like a tab). from the background script, it is hard (probably impossible) to know where this tab resides, so I have to deal with it in the experimental code during `QuickFolders.Interface.toggleToolbar()` 

background code:

```js
  function onOptionsTabActivated() {
    // tell experiment to make QuickFolders toolbar visible
    if (isDebug) console.log("QuickFolders Options tab is displayed. Sending message to experimental code...");
    messenger.Utilities.displayMainToolbar(true, true); // force toolbar visible = true, say we are in options tab = true.
  }

  messenger.tabs.onCreated.addListener(async (activeTab) => {
    try {
      if (isDebug) console.log("onCreated() - Initial tab:", activeTab);

      // Wait for the URL to be set and tab to complete loading
      const finalUrl = await checkTabStatus(activeTab.id);
      if (isDebug) console.log(`Tab fully loaded with URL: ${finalUrl}`);

      if (finalUrl.startsWith(getOptionsPageURL())) {
        // this can also be triggered by the dialog!! so it may affect the main window on the mail tab
        // which we don't want...
        onOptionsTabActivated();
      }
    } catch (error) {
      console.error("Error in tabs.onCreated listener:", error);
    }
  }); 
```

relevant code in toggleToolbar (optionsMode is only set:
```js
  let isPreviewMode = toggleOptions.optionsMode;
  if (
    QuickFolders.Interface.CurrentTabMode &&
    QuickFolders.Interface.CurrentTabMode.startsWith("mail")
  ) {
      // [issue 557]
      isPreviewMode = false;
  }
  QuickFolders.Interface.showElement(this.PreviewLabel, isPreviewMode);

```

I also see now that if the toolbar is _not_ displayed, opening the options window in the dialog may actually force the toolbar to become visible, so I still need to deal with this in a better way...

**Comment by RealRaven2000** [comment]
> I also see now that if the toolbar is _not_ displayed, opening the options window in the dialog may actually force the toolbar to become visible, so I still need to deal with this in a better way...

that may actually be a good thing, as we can see changes to the layout (bling my tabs page). So we can take this as a temporary win... let's say by switching the default behavior to displaying options in a tab  this may become less relevant anyway.

**Comment by RealRaven2000** [comment]
> I already hold a Pro license and renew annually without hesitation but would like to make an additional donation, please advise on how I can do so.

If you like, you can do it from here: https://quickfolders.org/donate.html

**Comment by RealRaven2000** [comment]
Fixed in 6.10.1. - Published 20/03/2025

**Comment by RealRaven2000** [comment]
Got a report from a user who still experience the issue in 136 / 128. Here is a fresh version for testing:

[QuickFolders-mx-6.10.4pre4.zip](https://github.com/user-attachments/files/19436740/QuickFolders-mx-6.10.4pre4.zip)

please enable Debug mode and `extensions.quickfolders.debug.interface = true` in QuickFolders Settings >> Advanced.

![Image](https://github.com/user-attachments/assets/bf553109-c463-4fcb-bb87-6d267d719f4a)

----

To try out this version, download the zip file and then drag it into Thunderbird Add-ons Manager (without extracting)
for users of screenreader you can also install this via the tools button after the "Find more addons" search text. activate the button with Enter then chose the 3rd menu entry "install addon from file...". Then select the zip file to install.

**Comment by RealRaven2000** [comment]
I am closing this for now unless somebody can reproduce it.

**Comment by LinuxOnTheDesktop** [comment]
If I may add this here: what is this 'preview mode' and why it is (-does it show as-) enabled when one . . merely opens the QF settings?

**Comment by RealRaven2000** [comment]
> If I may add this here: what is this 'preview mode' and why it is (-does it show as-) enabled when one . . merely opens the QF settings?

it just means that the main QF toolbar is showing while you are in settings. you may have noticed that settings in older versions were in a dialog so that you could observe any changes on the layout (pling my tabs) page could instantly be previewed. In order to see this while you are on the settings tab I had to enable the toolbar without it actually "working" there. That means, on the settings tab it doesn't really work the same as on any mail tab.
